configure

```diff
@@ -779,6 +779,8 @@ if try $CC -c $CFLAGS $test.c; then
   mv zconf${SUFFIX}.temp.h zconf${SUFFIX}.h
   echo "Checking for unistd.h... Yes." | tee -a configure.log
 else
+  sed < zconf${SUFFIX}.h "/^#ifdef HAVE_UNISTD_H.* may be/s/def HAVE_UNISTD_H\(.*\)/ ~ HAVE_UNISTD_H + 1\1/" > zconf${SUFFIX}.temp.h
```

Comment:
What is happening here?

Reply:
This is just as hard to parse as line 778 which I based it on. It does the same as cmake i.e when unistd.h is not available then replace `#ifdef HAVE_UNISTD_H` with `#ifdef ~ HAVE_UNISTD_H + 1`

Reply:
I see. So `#ifdef ~ HAVE_UNISTD_H + 1` cover the case where `HAVE_UNISTD_H` is not defined?

Reply:
It does work with all compilers actually as expected i.e the code block is not enabled. https://godbolt.org/z/afdqonqah